### DISCUSSION TITLE
wasRecentlyCreated is being ignored

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#124] [General]
Hello,

I am moving my Laravel app to use Laravel-data as DTO and replacing resources, too.

My tests just throw errors, as the API does no longer return HTTP 201 on store methods - I guess `wasRecentlyCreated` is being ignored? 

```
<?php

public function store(Request $request): ItemData {

    $item = Item::create([
                             'title' => $request->input('title'),
                         ]);

    return ItemData::from($item->loadMissing('location')); //returns HTTP 200 instead of HTTP 201
}
```

Is there any chance to get this working as expected?

**Comment by rubenvanassche** [comment]
Mhhh, I don't have a good solution in mind since anything could be passed to from not only models. And when returning the data object as a response we don't have a notion what was passed into the from method.

My only solution would be manually creating the response:

```php
response(ItemData::from($item->loadMissing('location')), 201);
```

**Comment by ahoiroman** [comment]
Closed with https://github.com/spatie/laravel-data/pull/291 - thanks!
